### PR TITLE
Bug 1159028 - [music] Refactor the music database backend (part 1)

### PR DESCRIPTION
**Pull request by jimporter**
WIP

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4c2fcc32c7529ce84b2ebc1cb3b309cad382bfb7

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4c2fcc32c7529ce84b2ebc1cb3b309cad382bfb7

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e9d37a6d85a6e6bc79b1a5797ae7ff61f5445d10

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4b1bee508467b39b7c0e25533a95cfde611c0596

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c8051261eace87eae1bdafc70557ca036f9b22a6

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c8867c08f62f6a29280f14b234aa7a0409a3f36d

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3b4568bc3e4a7aee155330c894f1920dbf8cb00e

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=268874fb0441f5dbf38102fabb44bdec6406cc3f

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cf473df71c8f0be66bef126f01162a7abeba7401

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5007954d3f9b304a022c4dbf2cca836cf1585561

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3796e3d221ea10ce90031f8a14eb61be1ef3a464

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2b826803e26317b75349a460b6198e5694638bbd

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d6ae650acf7085df7f712404f61ded1c51aa075d

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=64246fcf94791a47020eb4a7df726d1aea3f8c1d

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bd8c658c6e2c656956434ce3c40d98c66dcc9d77

File: apps/music/js/ui/views/search_view.js [hunk 1]
```diff
@@ -127,7 +127,7 @@ var SearchView = {
 
     if (this.searchContext === this.context.ALL ||
         this.searchContext === this.context.SONGS) {
-      this.searchHandles.title = musicdb.enumerate(
+      this.searchHandles.title = Database.enumerate(
         'metadata.title',
```
Comment (dominickuo) [inline apps/music/js/ui/views/search_view.js]: Looks like after we wrapped the MediaDB into the Database object, we need to fill the `range` and `direction` arguments or there will be an error says:

`[JavaScript Error: "DataError: Data provided to an operation does not meet requirements." {file: "app://music.gaiamobile.org/shared/js/mediadb.js" line: 1170}]`

when music is searching in all songs view.

File: apps/music/js/db.js [hunk 2]
```diff
@@ -1,244 +1,281 @@
...
-    // Stop and reset the player then back to tiles mode to avoid
-    // crash.  We could be smarter here by looking at the currently
-    // playing song and only stopping it if its volume is not in the
-    // list of available volumes. But that could potentially cause
-    // problems if we are playing a playlist and some songs are on one
-    // storage area and some in another. Yanking out an sdcard is
-    // uncommon enough that it should be fine to always stop playing.
-    if (typeof PlayerView !== 'undefined') {
-      PlayerView.stop();
+    var startedReparsing = false;
```
Comment (dominickuo) [inline apps/music/js/db.js]: Let's have an empty line below here.

File: apps/music/js/ui/views/search_view.js (hunk 1 above)
Comment (jimporter) [inline apps/music/js/ui/views/search_view.js]: Weird. I wonder why that's happening...

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b83619a353c996e5043bf551dcbd78d48ed89cbd

File: apps/music/js/ui/views/search_view.js (hunk 1 above)
Comment (jimporter) [inline apps/music/js/ui/views/search_view.js]: Ah ha. MediaDB checks arguments.length. I fixed this.

**Pull request merged**